.github/workflows/nightly-test.yml

```diff
@@ -0,0 +1,17 @@
+name: Gradle Nightly Test
+on:
+  schedule:
+    - cron: 0 0 * * * # daily
+jobs:
+  nightly-test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v3
+      with:
+        distribution: temurin
+        java-version: 17
+    - uses: gradle/gradle-build-action@v2
+      with:
+        gradle-version: nightly
```

Comment:
This wont run the intergration tests with the nightly gradle, will only try to build loom its self with it, somewhat useful to know, but not what we are going for.

Reply:
If you look below, it does run `gradle test` (which includes the integration tests), but without the matrix it'll take quite a long time

Reply:
Ah yeah, I forgot, this will run the test with the nightly gradle specified here, as well as the one in the test constants, only needs to run them on the "current" gradle version.

Reply:
I'll get round to this at some point, thanks

Reply:
Sounds good 👍 Let me know if you need some pointers, Gradle/Loom can be confusing :)